### ISSUE TITLE
[BUG]: Rule comment_012 : Keywords are detected even as part of an other word

### ISSUE THREAD

**Issue by wimille** [jeremiah-c-leary/vhdl-style-guide#1544]
**Environment**
version 3.35 on RHEL 8.10

**Describe the bug**
Rule comment_012 is activated and configured with specific keyword such as BUG and TODO.
However, this keyword are also detected as part of other words. For example, term BUG is detected if word DEBUG is used.

**To Reproduce**
Simple VHDL files (change extension to vhd or vhdl)
```
------------------------------------------------------------------------------------------------------------------------
-- Date         : 20/08/2020                                                                                          --
-- Version      : v1.00                                                                                               --
-- Auteur       :                                                                                                     --
-- Description  : Mise a jour du module:                                                                              --
--                  + Suppression de la partie DEBUG                                                                  --
--                  + Explication du terme BOBOTODOISE                                                                --
------------------------------------------------------------------------------------------------------------------------

library ieee;
use     ieee.std_logic_1164.all;
```

Simple VSG configuration file: 
```
rule:
  global:
    disable: true

  comment_012:
    disable: false
    keywords: ["TODO", "FIXME", "DEADCODE", "BUG"]
```

Commands:
`vsg -c comment_012.yaml -ap -f uart_rx_test.txt`

Actual result:

<img width="1095" height="232" alt="Image" src="https://github.com/user-attachments/assets/21e1a375-0042-4bb8-93ea-ff5bcd8a582d" />

**Expected behavior**
No violation should be detected as keywords are not present in the VHDL file.

**Screenshots**
N/A

**Additional context**
N/A

**Comment by JHertz5** [comment]
Hi @wimillie. Thanks for raising this issue. I can recreate your problem, and I'll look in to a solution.

**Comment by JHertz5** [comment]
Hi @wimille. I've made some changes on branch `issue-1544` that I believe will resolve the issue, and I've raise a PR (#1546). Please give the changes a try when you have a chance and let me know how it goes. If you're happy. I'll merge the PR.

**Comment by wimille** [comment]
Hi

I can't see the branch you talk about.

**Comment by JHertz5** [comment]
Apologies, I should have mentioned - it's on my fork, you should be able to see it if you look at the PR

**Comment by wimille** [comment]
Sorry for my remark, i'm not very familiar with github
However here's my test:
<img width="1109" height="565" alt="Image" src="https://github.com/user-attachments/assets/6a20aca3-c5e6-4b4d-949a-82172491fc0e" />

Unfortunately i've got the same warning.
But to be honnest i've a doubt about my command. I'm wondering if i use vsg version from your repository or the version i've got on my station.

**Comment by wimille** [comment]
Welll as i said i was not sure to use the correct vsg version and correct rule files. 
So  i've made a very dirty thing, i've modified the rule_012.py file on my station. Since this i could made the test and no warnings are raised. Until i make some additionnal test.


```
vsg -rc comment_012 -c bin/commen_012.yaml
{
  "rule": {
    "comment_012": {
      "indent_style": "spaces",
      "indent_size": 2,
      "phase": 1,
      "disable": false,
      "fixable": false,
      "severity": "Warning",
      "user_error_message": "",
      "keywords": [
        "TODO",
        "FIXME",
        "DEADCODE",
        "BUG"
      ]
    }
  }
}
```

And my VHD file:
```
------------------------------------------------------------------------------------------------------------------------
-- Date         : 20/08/2020                                                                                          --
-- Version      : v1.00                                                                                               --
-- Auteur       :                                                                                                     --
-- Description  : Mise a jour du module:                                                                              --
--                  + Suppression de la partie DEBUG                                                                  --
--                  + Explication du terme BOBOTODOISE                                                                --
-- DEADCODE
------------------------------------------------------------------------------------------------------------------------


library ieee;
use     ieee.std_logic_1164.all;


-- SOME CRAZY THINGS


entity DEBUG_TODO is
port(
    TODO : in std_logic;  -- DEADCODE
    FIXME : in std_logic;
    DEBUG : out std_logic
);
end entity;

architecture rtl of DEBUG_TODO is

signal todo_3 : std_logic;
    signal bob_todo : std_logic;

begin
    --FIXME
    todo_3 <= TODO xor '1';
    bob_todo <= todo_3 after 2 ns;

    DEBUG <= bob_todo nand FIXME when todo_3 else '0';

end architecture;
```
 Results was not the expected one

<img width="1106" height="306" alt="Image" src="https://github.com/user-attachments/assets/6a762dca-c24b-42cf-a24d-483d5bbf53f9" />

**Comment by wimille** [comment]
Hi
I want to apologize, i've made a mistake in my previous message. The rule configuration "keywords" is used to set the words you want to detect, and i though it was the keyword you want to except. 
So it is normal that in my last example, the 3 warnings are well detected.

Your correction works fine, thank you.

**Comment by JHertz5** [comment]
Hi @wimillie

Apologies, things have been busy for me at work so I didnt get a chance to investigate your messages. Thanks for the update, I'm glad you're happy with the fix! I'll merge the PR and it'll be part of the next release.
